src/lib/server/query_handler.cpp

```diff
@@ -96,7 +96,7 @@ void QueryHandler::handle_transaction_statement_message(ExecutionInformation& ex
   auto& statements = sql_statement->getStatements();
 
   if (statements.front()->isType(hsql::StatementType::kStmtTransaction)) {
-    auto* transaction_statement = reinterpret_cast<hsql::TransactionStatement*>(statements.front());
+    auto* transaction_statement = dynamic_cast<hsql::TransactionStatement*>(statements.front());
```

Comment:
Making this its own comment so that it does not get lost in the other discussion:

> Also, one more thing: The hsql namespace should not make it outside of the SQLTranslator. Could you please translate kCommitTransaction etc. into a Hyrise-internal symbol? That improves the encapsulation.

Reply:
Obsolete (as hsql is already used in the pipeline).